citesphere-importer/src/main/java/edu/asu/diging/citesphere/importer/core/service/parse/iterators/CrossRefIterator.java

```diff
@@ -0,0 +1,279 @@
+        typeMap.put("proceedings-article", Publication.PROCEEDINGS_ARTICLE);
+        typeMap.put("dissertation", Publication.DISSERTATION);
+        typeMap.put("book-section", Publication.BOOK_SECTION);
+        typeMap.put("report-component", Publication.REPORT_COMPONENT);
+        typeMap.put("report", Publication.REPORT);
+        typeMap.put("peer-review", Publication.PEER_REVIEW);
+        typeMap.put("book-track", Publication.BOOK_TRACK);
+        typeMap.put("book-part", Publication.BOOK_PART);
+        typeMap.put("other", Publication.OTHER);
+        typeMap.put("journal-volume", Publication.JORUNAL_VOLUME);
+        typeMap.put("book-set", Publication.BOOK_SET);
+        typeMap.put("journal", Publication.JOURNAL);
+        typeMap.put("proceedings-series", Publication.PROCEEDINGS_SERIES);
+        typeMap.put("report-series", Publication.REPORT_SERIES);
+        typeMap.put("proceedings", Publication.PROCEEDINGS);
+        typeMap.put("database", Publication.DATABASE);
+        typeMap.put("standard", Publication.STANDARD);
+        typeMap.put("reference-book", Publication.REFERENCE_BOOK);
+        typeMap.put("grant", Publication.GRANT);
+        typeMap.put("dataset", Publication.DATASET);
+        typeMap.put("book-series", Publication.BOOK_SERIES);
+        parseCrossRef();
+
+    }
+
+    private void parseCrossRef() {
+
+        articles = new ArrayList<>();
+        for (String doi : info.getDois()) {
+            try {
+                Item item = crossrefService.get(doi);
+                article = new Publication();
+                article.setArticleType(typeMap.get(item.getType())); 
+                article.setJournalMeta(parseJournalMeta(item));
+                article.setArticleMeta(parseArticleMeta(item));
+
+                articles.add(article);
+            } catch (RequestFailedException | IOException e) {
+                logger.error("Couuld not retrieve work for doi: "+ doi, e);
+                // for now we just log the exceptions
+                // we might want to devise a way to decide if the 
+                // service might be down and we should stop sending requests.
+            }
+        }
+    }
+
+    private ContainerMeta parseJournalMeta(Item item) {
+        ContainerMeta meta = new ContainerMeta();
+        meta.setContainerTitle(String.join(", ", item.getContainerTitle()));
+        meta.setPublisherName(item.getPublisher());
+        meta.setPublisherLocation(item.getPublisherLocation());
+        List<Issn> issnList = new ArrayList<Issn>();
+        if(item.getIssnType() != null) {
+            for(IssnType issnType : item.getIssnType()) {
+                Issn issn = new Issn();
+                issn.setIssn(issnType.getValue());
+                issn.setPubType(issnType.getType());
+                issnList.add(issn);
+            }
+        }
+        meta.setIssns(issnList);
+
+        List<Contributor> contributors = new ArrayList<>();
+        if(item.getChair() != null) {
```

Comment:
is that the chair of what? the journal? or is that for theses/disserations/etc.? In that case, this is not journal metadata.

Reply:
Chair is usually the same for entire journal, as far as I am aware. That's why it's added to journal metadata.

Reply:
I don't think that's true. This would be filled for dissertations for example, or maybe for proceedings or something like that (e.g. https://api.crossref.org/works?query.chair=smith). I would keep it with the item metadata.

Reply:
?